msgs/proto/posts.proto

```diff
@@ -0,0 +1,120 @@
+syntax = "proto3";
+
+// openboard
+//
+// user semantics:
+//
+//  Add{T}(s)    (Add{T}(s)Req)    returns {T}(s)Resp // POST
+//  Ovr{T}(s)    (Ovr{T}(s)Req)    returns {T}(s)Resp // PUT
+//  Mod{T}(s)    (Mod{T}(s)Req)    returns {T}(s)Resp // PATCH
+//  Get{T}       (Get{T}Req)       returns {T}Resp    // GET
+//  Fnd{T}s      (Fnd{T}sReq)      returns {T}sResp   // GET
+//  Rmv{T}(s)    (Rmv{T}(s)Req)    returns EmptyResp  // DELETE
+//  Unr{T}       (Unr{T}Req)       returns {T}Resp    // PATCH
+package pb;
+
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+
+message CommentResp {
+
+}
+
+message AddCommentResp {
+
+}
+
+message TypeResp {
+	uint32 id = 1;
+	string name = 2;
+}
+
+message AddTypeReq {
+	string name = 1;
+}
+
+message TypesResp {
+	repeated TypeResp items = 1;
+	uint32 total = 2;
+}
+
+message FndTypesReq {
+	uint32 limit = 1;
+	uint32 lapse = 2;
+}
+
+message PostResp {
+	uint32 id = 1;
+	string slug = 2;
+	string title = 3;
+	string body = 4;
+	string typeId = 5;
+	google.protobuf.Timestamp created = 6;
+    google.protobuf.Timestamp updated = 7;
+    google.protobuf.Timestamp deleted = 8;
+    google.protobuf.Timestamp blocked = 9;
+}
+
+message AddPostReq {
+	string title = 1;
+	string body = 2;
+	uint32 typeId = 3;
```

Comment:
Would the front end provide the `slug` for adding posts as well? Or is that decided by the backend?

Reply:
This was clarified in-person.